cleaner/email/classes/clean.php

```diff
@@ -0,0 +1,177 @@
+
+        self::execute_configreplace($config, $verbose, $dryrun);
+        self::next_step();
+
+        self::execute_appendsuffix($config, $verbose, $dryrun);
+        self::next_step();
+
+        self::execute_ignoresuffix($config, $verbose, $dryrun);
+        self::next_step();
+    }
+
+    /**
+     * Replace the defined $CFG settings.
+     *
+     * @param stdClass $config
+     * @param bool $verbose
+     * @param bool $dryrun
+     *
+     */
+    public static function execute_configreplace($config, $verbose, $dryrun) {
+        $cfgsettings = [
+            'noemailever',
+            'divertallemailsto',
+            'divertallemailsexcept',
+        ];
+
+        foreach ($cfgsettings as $setting) {
+            $value = $config->$setting;
+
+            if ($verbose) {
+                mtrace("Executing: set_config('$setting', ********);");
+            }
+
+            if (!$dryrun) {
+                set_config($setting, $value);
+            }
+        }
+    }
+
+    /**
+     * Append a suffix to all {user} email addresses.
+     *
+     * @param stdClass $config
+     * @param bool $verbose
+     * @param bool $dryrun
+     *
+     * @return bool
+     */
+    public static function execute_appendsuffix($config, $verbose, $dryrun) {
+        global $DB;
+
+        $dbtype = $DB->get_dbfamily();
+        $suffix = $config->emailsuffix;
+
+        $query = '';
+
+        if (empty($suffix)) {
+            return false;
+        }
+
+        if ($dbtype == 'postgres') {
+            $query = "UPDATE {user} SET email = email || '$suffix'";
+        } else if ($dbtype == 'mysql') {
+            $query = "UPDATE {user} SET email = CONCAT(email, '$suffix') ";
+        } else {
+            mtrace("Database not supported: $dbtype");
+        }
+
+        if ($verbose) {
+            mtrace("Executing: $query");
+        }
+
+        if (!$dryrun) {
+            $DB->execute($query);
+        }
+
+        return true;
+    }
+
+    /**
+     * Remove the suffix from a set of users based on the defined regular expression.
+     *
+     * @param stdClass $config
+     * @param bool $verbose
+     * @param bool $dryrun
+     *
+     * @return bool
+     */
+    public static function execute_ignoresuffix($config, $verbose, $dryrun) {
+        global $DB;
+
+        $dbtype = $DB->get_dbfamily();
+        $suffix = $config->emailsuffix;
+        $emailsuffixignore = $config->emailsuffixignore;
+
+        $query = '';
+
+        if (empty($suffix) || empty($emailsuffixignore)) {
+            return false;
+        }
+
+        if ($dbtype == 'postgres') {
+            $query = "UPDATE {user} SET email = regexp_replace(email, '$suffix', '', 'g') WHERE email ~ '$emailsuffixignore'";
+        } else if ($dbtype == 'mysql') {
+            $query = "UPDATE {user} SET email = TRIM(TRAILING '$suffix' FROM email) WHERE email REGEXP '$emailsuffixignore'";
+        } else {
+            mtrace("Database not supported: $dbtype");
+        }
```

Comment:
I have removed that function and added a inverse regex to the first append function. $DB->sql_regex(false)